### ISSUE TITLE
Plan for SegWit Support?

### ISSUE THREAD

**Issue by KeyJockey** [bitpay/copay#3844]
Guys, might you want to add Copay wallet to this list here re: Segregated Witness support?

(That is, assuming you DO plan to support it?  OR if not, why not?  I'd be interested in hearing opinion on it here pro, or con... either way?) 

https://bitcoincore.org/en/segwit_adoption/

**Comment by martindale** [comment]
+1 on SegWit adoption, especially if it leads to official support in Bitcore.

**Comment by dabura667** [comment]
Segregated Witness support is not that difficult. And there's no reason to not support it if it gains massive adoption.

The list is merely a political move.

IMO Copay should support segwit but should not put their name on that list.

**Comment by KeyJockey** [comment]
LOL man, well srsly what's NOT "political" in bitcoin these days?

Unfortunately there is no getting around social/politics stuff, as much as we would like to avoid it.  And if SegWit isn't difficult and it's likely to be supported easily, then why NOT add Copay to this "political list" anyway?  NOT having our favorite wallet (and IMHO the best one) listed here on this stupid page just makes Copay look bad, somehow, compared to all the other wallets who are listing themselves.

It makes it look to unaware users that somehow something is "wrong" with Copay if it's not there.  Doesn't avoid the politics unfortunately just reflects badly on Copay :frowning:

**Comment by miguelmorales85** [comment]
Looking for Copay was the first thing I did and I did not found them... :-/

**Comment by matiu** [comment]
Yes, we are planning to support segwit. https://github.com/bitpay/bitcore-lib/pull/50

**Comment by KeyJockey** [comment]
OK then... so, why not add Copay to this dumb bitcoincore list then?  

I still maintain that Copay NOT being listed here only reflects badly on Copay (for people unaware of the larger issues, that is... )

All that being said, FWIW I'm actually not a "core" supporter myself at this point - lately reading all the back and forth B.S. I've been coming down more on the side of the "classic" POV in the whole block size mess: as a "simple end user" not programmer I *really hope* that this whole clusterf*ck issue IS RESOLVED SOON so that Bitcoin can get back to being BITCOIN :)

**Comment by matiu** [comment]
> OK then... so, why not add Copay to this dumb bitcoincore list then?

sure, you are welcome to submit it! Thanks.

**Comment by martindale** [comment]
@matiu can you please re-open this issue and only close it once Segregated Witness is indeed supported within Copay, not just Bitcore?

**Comment by matiu** [comment]
@martindale I closed because the issue was "plan to support segwit" :)  We have a plan already: Bitcore -> Insight -> BWS -> Copay.  We are at the first fase too. I created:https://github.com/bitpay/copay/issues/4020

thanks!

**Comment by matiu** [comment]
Thanks @miguelmorales85

**Comment by martindale** [comment]
Thanks!  In the future, you may want to keep the original requests intact and simply re-name the issue (i.e., groom the backlog) in a way that makes sense in the context of the project.  This allows the submitter (generally, the one most interested in the feature) to be notified of progress over time, and preserves the temporal relevance of the initial inquiry.

**Comment by entreprenewer** [comment]
A year later, a month after 2X failed, and still no SegWit support. Meanwhile, BitPay's founder is complaining about high transaction fees on Twitter - and their app defines "super economy" as 128sat/byte... No joke, guys. Add SegWit.

Clearly political. What a shame.

**Comment by junderw** [comment]
@entreprenewer There was a post in another issue where it was shown:

1. public facing insight was using old bitcoind, so estimatefee was 2x higher than necessary. (since it didn't see the witness data at all, all witness transactions would look like they were paying high satoshi/byte fees (as bytes appear lower than weight based calculations))
2. public facing insight still uses estimatefee instead of estimatesmartfee which is much more robust. This could lower fees another 30%.
3. Segwit support at the BWS / Copay level would require some UI work... but once this is done transaction sizes will go down by about 40% for single sig and 50-60% for multisig lowering absolute fees by that much.

Currently they have segwit support in bitcore-lib and they switched their insight backend to a segwit-capable recent version.

All they need is to implement into BWS and BWC and then Copay's UI needs adjusting (but I guess they could try and reuse some of the BTC/BCH separating UI concepts. Make BTC wallets called Legacy and Segwit BTC wallets become the new "Bitcoin Wallets" would be cool. Similar to Trezor's UI.

**Comment by miguelmorales85** [comment]
It's been very well know that Bitpay follows another agenda.. I think they will delay Segwit until it is completely necessary. Remember how they focus they development to integrate Bcash to Copay even when there was a functional tool to do it. I will move on to Samourai Wallet. It was good while it lasted.
Peace

**Comment by markasoftware** [comment]
Is Bitpay opposed to adding SegWit support, or just nobody has stepped up to develop it? If somebody were to add support, would they accept the PR?
